app/src/main/java/org/xwiki/android/sync/contactdb/User.kt

```diff
@@ -0,0 +1,15 @@
+package org.xwiki.android.sync.contactdb
+
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Entity (tableName = "user_table")
+data class User (
+    @PrimaryKey var uid: String,
+    @ColumnInfo(name = "account_name") val accountName: String,
```

Comment:
As I have seen previously, you are using account name and server address for generating of unique id. So, it is better to use combination of two primary fields - account name and server address. In this case their combination will be guarantee unique.

Reply:
Like this? 
@Entity (tableName = "user_table", primaryKeys = ["account_name", "server_address"])